hera_mc/cm_redis_corr.py

```diff
@@ -72,25 +36,33 @@ def cminfo_redis_snap(cminfo, redis_info=None):
     snap_to_ant = {}
     ant_to_snap = {}
     all_snap_inputs = {}
-    for antn, ant in enumerate(cminfo['antenna_numbers']):
-        name = cminfo['antenna_names'][antn]
-        pol_info = {}
-        for pol_snap in cminfo['correlator_inputs'][antn]:
-            pol = pol_snap.lower()[0]
-            if pol in ['e', 'n'] and '>' in pol_snap:
-                pol_info[pol] = pol_snap
+    snap_to_serial = {}
+    for _n, ant in enumerate(cminfo['antenna_numbers']):
+        name = cminfo['antenna_names'][_n]
+        corr_inp = cminfo['correlator_inputs'][_n]
+        snap_snr = cminfo['snap_serial_numbers'][_n]
         ant_to_snap[ant] = {}
-        for pol, psnap in pol_info.items():
-            snapi, channel = snap_part_to_host_input(psnap, redis_info=redis_info)
-            ant_to_snap[ant][pol] = {'host': snapi, 'channel': channel}
-            snap_to_ant.setdefault(snapi, [None] * 6)
-            snap_to_ant[snapi][channel] = name + pol.upper()
-            snap, adc_num = snap_part_to_host_input(psnap, None)
-            all_snap_inputs.setdefault(snap, [])
-            all_snap_inputs[snap].append(adc_num)
+        for _i, psnap in enumerate(corr_inp):
+            pol = psnap.lower()[0]
+            if pol not in ['e', 'n'] or '>' not in psnap:
+                warnings.warn(f"{psnap} is not an allowed correlator input")
+                continue
+            snap_input, snap_hostname = psnap.split('>')
+            channel = int(snap_input[1:]) // 2  # divide by 2 because ADC is in demux 2
+            ant_to_snap[ant][pol] = {'host': snap_hostname, 'channel': channel}
+            snap_to_ant.setdefault(snap_hostname, [None] * 6)
+            snap_to_ant[snap_hostname][channel] = name + pol.upper()
+            all_snap_inputs.setdefault(snap_hostname, [])
+            all_snap_inputs[snap_hostname].append(snap_input)
+            try:  # if present, check if hostnames are the same between polarizations.
+                if snap_to_serial[snap_hostname] != snap_snr[_i]:
+                    warnings.warn(f"snap_hostname {snap_hostname} differs between inputs")
+            except KeyError:  # if not present, add it
+                snap_to_serial[snap_hostname] = snap_snr[_i]
```

Comment:
Looks like now this does not overwrite if they are different. Does this matter which one is tracked?

Reply:
I now correctly raise an error if the hostname to serial numbers are not consistent.